File: examples/swift-client/Sources/AppwriteModels/Preferences.swift

/// Preferences
public class Preferences {

    let data: [String: Any]

    init(
        data: [String: Any]
    ) {
        self.data = data
    }

    public static func from(map: [String: Any]) -> Preferences {
        return Preferences(
            data: map
        )
    }

    public func toMap() -> [String: Any] {
        return [
            "data": data
        ]
    }

    public func convertTo<T>(fromJson: ([String: Any]) -> T) -> T {
        return fromJson(data)
    }
        
}

File: examples/flutter/lib/src/models/preferences.dart
part of appwrite.models;

/// Preferences
class Preferences {
    final Map<String, dynamic> data;

    Preferences({
        required this.data,
    });

    factory Preferences.fromMap(Map<String, dynamic> map) {
        return Preferences(
            data: map,
        );
    }

    Map<String, dynamic> toMap() {
        return {
            "data": data,
        };
    }

    T convertTo<T>(T Function(Map) fromJson) => fromJson(data);
}
